src/fides/api/alembic/migrations/versions/c9c72b3d550b_data_migration_converting_existing_.py

```diff
@@ -0,0 +1,214 @@
+"""data migration converting existing 'Cookies' resources to 'Asset' resources with type 'Cookie'
+
+Revision ID: c9c72b3d550b
+Revises: c961528edfc6
+Create Date: 2025-02-18 18:33:56.039924
+
+"""
+
+import uuid
+from typing import Dict
+
+import psycopg2
+import sqlalchemy as sa
+from alembic import op
+from loguru import logger
+
+from fides.api.models.asset import Asset
+
+# revision identifiers, used by Alembic.
+revision = "c9c72b3d550b"
+down_revision = "c961528edfc6"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    try:
+        op.add_column("asset", sa.Column("description", sa.String(), nullable=True))
+    except psycopg2.errors.DuplicateColumn:
+        logger.warning(
+            "Column 'description' already exists in 'asset' table. Skipping column addition."
+        )
+
+    connection = op.get_bind()
+    result = connection.execute(
+        sa.text(
+            "SELECT id, created_at, updated_at, name, domain, system_id, path, privacy_declaration_id FROM cookies"
+        )
+    )
+
+    logger.debug("Converting existing cookies to assets")
+    assets_to_create: Dict[str, Asset] = {}
+    for row in result:
+        if row.privacy_declaration_id is None:
+            # If the privacy declaration ID is None, we skip matching for this row
+            assets_to_create[row.name] = Asset(
+                id=str(uuid.uuid4()),
+                created_at=row.created_at,
+                updated_at=row.updated_at,
+                name=row.name,
+                domain=row.domain,
+                system_id=row.system_id,
+                data_uses=[],
+                asset_type="Cookie",
+            )
+            continue
+        pud_system_id_data_use = connection.execute(
+            sa.text(
+                "SELECT system_id, data_use FROM privacydeclaration WHERE id = :privacy_declaration_id"
+            ),
+            privacy_declaration_id=row.privacy_declaration_id,
+        ).fetchone()
```

Comment:
ideally we'd pull this data in with our original cookie query in an outer join on the `privacy_declaration_id`? as it stands now, i think this presents an an n+1 query pattern. the performance implications for a migration are obviously not as important as for normal app runtime, but in the past we've had migrations time out, and it's not inconceivable that someone could have tens of thousands of cookies on their server (which would result in tens of thousands of queries being executed here). so i do think we should try to optimize this

Reply:
removed the n+1 queries, bulk fetch, bulk insert for upgrade/downgrade